modin/pandas/base.py

```diff
@@ -2532,13 +2532,9 @@ def _set_axis(self, labels, axis, inplace, copy):  # noqa: PR01, RT01, D200
                 stacklevel=2,
             )
             labels, axis = axis, labels
-        if inplace:
-            setattr(self, pandas.DataFrame()._get_axis_name(axis), labels)
-        else:
-            if copy:
-                obj = self.copy()
-            obj.set_axis(labels, axis=axis, inplace=True)
-            return obj
+        obj = self.copy() if copy and inplace == False else self
```

Comment:
I'm still wondering what's the correct behaviour when `copy=True` and `inplace=True`... technically they're about different things (`copy` is about copying vs not copying underlying data, like shallow vs deep copy, and `inplace` is for operation itself), so I'm not entirely sure that your current line is correct.

Though I would still consider it semantically correct, because for me both set to `True` is meaningless.

Reply:
pandas raise exception in that case. Updated our code accordingly.